### DISCUSSION TITLE
is it possible to define the interceptor when using curl-cffi

### DISCUSSION THREAD

**Discussion by jiangxiaoqiang** [lexiforest/curl_cffi#153] [Q&A]
I am using curl-cffi as the client of http request, Now  I facing a isssue that I want to check the JWT access token expire or not before send the http request, is is possible to define the interceptor when using curl-cffi? I have already read the document but did not find the introduce of the interceptor.

**Comment by perkfly** [comment]
Can you give an example of interceptors? The only interceptor I've heard of is `axios`'s.

**Reply by jiangxiaoqiang** [reply]
seems someone tried to implement this feature: https://github.com/manmolecular/py-request-interceptor, I also only fimiliar with the  axios intercepter and not fimiliar with python request lib. I am not sure it should be implement by curl-cffi. I have to check the token expire in every function and I think the code check token expire in each request function is very ugly.
